subsys/usb/usb_device.c

```diff
@@ -1437,13 +1437,23 @@ static int custom_handler(struct usb_setup_packet *pSetup,
 			continue;
 		}
 
-		/* An exception for AUDIO_CLASS is temporary and shall not be
-		 * considered as valid solution for other classes.
-		 */
-		if (iface->custom_handler &&
-		    (if_descr->bInterfaceNumber == (pSetup->wIndex & 0xFF) ||
-		     if_descr->bInterfaceClass == AUDIO_CLASS)) {
+		if (iface->custom_handler == NULL) {
+			continue;
+		}
+
+		if (if_descr->bInterfaceNumber == (pSetup->wIndex & 0xFF)) {
 			return iface->custom_handler(pSetup, len, data);
+		} else {
```

Comment:
you can concatenate this to single `else if` instead. No need to add braces here imho.